### DISCUSSION TITLE
After 25min  of running thread consumer no consume anything!

### DISCUSSION THREAD

**Discussion by mmdanny89** [confluentinc/confluent-kafka-python#1724] [Q&A]
```
c = Consumer({
    'bootstrap.servers': 'localhost',
    'group.id': 'tests_x',
    'auto.offset.reset': 'earliest'
})
consumer.subscribe(['example'])
def _consume_data(consumer):
    while True:
        msg = consumer.poll(1.0)
        if msg is None:
            continue
        if msg.error():
            print("Consumer error: {}".format(msg.error()))
            continue
        print('Received message: {}'.format(msg.value().decode('utf-8')))
    consumer.close()
```

in this example after 25-30 min consumer no consume messages any help with this

**Comment by abdelghanimeliani** [comment]
hey @mmdanny89 , I haven't been using this library for long, but I'm encountering similar issues. It might be that you're trying to read events from the beginning of the topic, and there are already a considerable number of events. If this isn't necessary, try setting 'auto.offset.reset' to 'latest'. Also, ensure that you are using the correct topic names.

**Reply by mmdanny89** [reply]
Thanks, already make a solution for this, and work fine. have a nice week.
